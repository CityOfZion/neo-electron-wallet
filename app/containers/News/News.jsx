// @flow
import React from 'react'

type Props = {
  feed: {
    items: Array<{
      content: String,
      creator: String,
    }>,
  },
}

export default class News extends React.Component<Props> {
  render() {
    return this.parseItems()
  }

  parseItems() {
    const { items } = this.props.feed
    return items.map(item => <div>{item.creator}</div>)
  }
}
