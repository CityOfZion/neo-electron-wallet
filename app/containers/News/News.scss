.newsPanel {
  min-height: 100%;
}

.newsContainer {
  height: 100%;
  display: flex;
  flex-direction: column;
}
