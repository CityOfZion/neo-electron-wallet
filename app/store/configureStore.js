import { createStore, applyMiddleware } from 'redux';
import { createLogger } from 'redux-logger';
import thunk from 'redux-thunk';
import rootReducer from '../reducers/rootReducer';
// export type GlobalState = {}
export var INITIAL_STATE = {};
var loggerMiddleware = createLogger();
function configureStore(initialState) {
    if (initialState === void 0) { initialState = INITIAL_STATE; }
    return createStore(rootReducer, initialState, process.env.NODE_ENV === 'production'
        ? applyMiddleware(thunk)
        : applyMiddleware(thunk, loggerMiddleware));
}
export var store = configureStore();
