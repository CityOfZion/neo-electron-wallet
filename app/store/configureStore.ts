import { createStore, applyMiddleware } from 'redux'
import { createLogger } from 'redux-logger'
import thunk from 'redux-thunk'
import rootReducer from '../reducers/rootReducer'

// export type GlobalState = {}
export const INITIAL_STATE = {}

const loggerMiddleware = createLogger()

function configureStore(initialState = INITIAL_STATE) {
  return createStore(
    rootReducer,
    initialState,
    process.env.NODE_ENV === 'production'
      ? applyMiddleware(thunk)
      : applyMiddleware(thunk, loggerMiddleware),
  )
}
export const store = configureStore()
