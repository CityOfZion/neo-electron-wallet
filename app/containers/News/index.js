// @flow
import { compose } from 'recompose'
import { withCall, withData } from 'spunky'

import withLoadingProp from '../../hocs/withLoadingProp'
import newsActions from '../../actions/newsActions'
import News from './News'

export default compose(
  withCall(newsActions),
  withData(newsActions),
  withLoadingProp(newsActions),
)(News)
