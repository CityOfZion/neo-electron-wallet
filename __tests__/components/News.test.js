import React from 'react'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import { MemoryRouter } from 'react-router-dom'
import { shallow, mount } from 'enzyme'

import News from '../../app/containers/News'

const setup = (
  shallowRender = true,
  state = {
    account: {
      loggedIn: true,
      wif: undefined,
    },
  },
) => {
  const store = configureStore()(state)
  const Component = News

  const wrapper = mount(
    <Provider store={store}>
      <MemoryRouter initialEntries={['/']} keyLength={0}>
        <Component store={store} />
      </MemoryRouter>
    </Provider>,
  )

  return {
    store,
    wrapper,
  }
}

describe('News', () => {
  test('renders without crashing', () => {
    const { wrapper } = setup()
    expect(wrapper).toMatchSnapshot()
  })
})
