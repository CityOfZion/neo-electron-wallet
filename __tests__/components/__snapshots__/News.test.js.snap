// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News renders without crashing 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MemoryRouter
    initialEntries={
      Array [
        "/",
      ]
    }
    keyLength={0}
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(withCall(Connect(withData(Connect(withProgress(withProps(Connect(withData(withProps(News))))))))))
        store={
          Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
        }
      >
        <withCall(Connect(withData(Connect(withProgress(withProps(Connect(withData(withProps(News)))))))))
          performAction={[Function]}
          store={
            Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            }
          }
        >
          <withoutProps(Connect(withData(Connect(withProgress(withProps(Connect(withData(withProps(News)))))))))
            performAction={[Function]}
            store={
              Object {
                "clearActions": [Function],
                "dispatch": [Function],
                "getActions": [Function],
                "getState": [Function],
                "replaceReducer": [Function],
                "subscribe": [Function],
              }
            }
          >
            <Connect(withData(Connect(withProgress(withProps(Connect(withData(withProps(News))))))))
              store={
                Object {
                  "clearActions": [Function],
                  "dispatch": [Function],
                  "getActions": [Function],
                  "getState": [Function],
                  "replaceReducer": [Function],
                  "subscribe": [Function],
                }
              }
            >
              <withData(Connect(withProgress(withProps(Connect(withData(withProps(News)))))))
                dispatch={[Function]}
                store={
                  Object {
                    "clearActions": [Function],
                    "dispatch": [Function],
                    "getActions": [Function],
                    "getState": [Function],
                    "replaceReducer": [Function],
                    "subscribe": [Function],
                  }
                }
                storeSubscription={
                  t {
                    "listeners": Object {
                      "clear": [Function],
                      "get": [Function],
                      "notify": [Function],
                      "subscribe": [Function],
                    },
                    "onStateChange": [Function],
                    "parentSub": undefined,
                    "store": Object {
                      "clearActions": [Function],
                      "dispatch": [Function],
                      "getActions": [Function],
                      "getState": [Function],
                      "replaceReducer": [Function],
                      "subscribe": [Function],
                    },
                    "unsubscribe": [Function],
                  }
                }
              >
                <withProgress(withProps(Connect(withData(withProps(News)))))
                  __progress__="LOADING"
                  dispatch={[Function]}
                  store={
                    Object {
                      "clearActions": [Function],
                      "dispatch": [Function],
                      "getActions": [Function],
                      "getState": [Function],
                      "replaceReducer": [Function],
                      "subscribe": [Function],
                    }
                  }
                  storeSubscription={
                    t {
                      "listeners": Object {
                        "clear": [Function],
                        "get": [Function],
                        "notify": [Function],
                        "subscribe": [Function],
                      },
                      "onStateChange": [Function],
                      "parentSub": t {
                        "listeners": Object {
                          "clear": [Function],
                          "get": [Function],
                          "notify": [Function],
                          "subscribe": [Function],
                        },
                        "onStateChange": [Function],
                        "parentSub": undefined,
                        "store": Object {
                          "clearActions": [Function],
                          "dispatch": [Function],
                          "getActions": [Function],
                          "getState": [Function],
                          "replaceReducer": [Function],
                          "subscribe": [Function],
                        },
                        "unsubscribe": [Function],
                      },
                      "store": Object {
                        "clearActions": [Function],
                        "dispatch": [Function],
                        "getActions": [Function],
                        "getState": [Function],
                        "replaceReducer": [Function],
                        "subscribe": [Function],
                      },
                      "unsubscribe": [Function],
                    }
                  }
                >
                  <Connect(withData(withProps(News)))
                    __progress__="LOADING"
                    dispatch={[Function]}
                    loading={true}
                    store={
                      Object {
                        "clearActions": [Function],
                        "dispatch": [Function],
                        "getActions": [Function],
                        "getState": [Function],
                        "replaceReducer": [Function],
                        "subscribe": [Function],
                      }
                    }
                    storeSubscription={
                      t {
                        "listeners": Object {
                          "clear": [Function],
                          "get": [Function],
                          "notify": [Function],
                          "subscribe": [Function],
                        },
                        "onStateChange": [Function],
                        "parentSub": t {
                          "listeners": Object {
                            "clear": [Function],
                            "get": [Function],
                            "notify": [Function],
                            "subscribe": [Function],
                          },
                          "onStateChange": [Function],
                          "parentSub": undefined,
                          "store": Object {
                            "clearActions": [Function],
                            "dispatch": [Function],
                            "getActions": [Function],
                            "getState": [Function],
                            "replaceReducer": [Function],
                            "subscribe": [Function],
                          },
                          "unsubscribe": [Function],
                        },
                        "store": Object {
                          "clearActions": [Function],
                          "dispatch": [Function],
                          "getActions": [Function],
                          "getState": [Function],
                          "replaceReducer": [Function],
                          "subscribe": [Function],
                        },
                        "unsubscribe": [Function],
                      }
                    }
                  >
                    <withData(withProps(News))
                      __progress__="LOADING"
                      dispatch={[Function]}
                      loading={true}
                      networkId={null}
                      store={
                        Object {
                          "clearActions": [Function],
                          "dispatch": [Function],
                          "getActions": [Function],
                          "getState": [Function],
                          "replaceReducer": [Function],
                          "subscribe": [Function],
                        }
                      }
                      storeSubscription={
                        t {
                          "listeners": Object {
                            "clear": [Function],
                            "get": [Function],
                            "notify": [Function],
                            "subscribe": [Function],
                          },
                          "onStateChange": [Function],
                          "parentSub": t {
                            "listeners": Object {
                              "clear": [Function],
                              "get": [Function],
                              "notify": [Function],
                              "subscribe": [Function],
                            },
                            "onStateChange": [Function],
                            "parentSub": t {
                              "listeners": Object {
                                "clear": [Function],
                                "get": [Function],
                                "notify": [Function],
                                "subscribe": [Function],
                              },
                              "onStateChange": [Function],
                              "parentSub": undefined,
                              "store": Object {
                                "clearActions": [Function],
                                "dispatch": [Function],
                                "getActions": [Function],
                                "getState": [Function],
                                "replaceReducer": [Function],
                                "subscribe": [Function],
                              },
                              "unsubscribe": [Function],
                            },
                            "store": Object {
                              "clearActions": [Function],
                              "dispatch": [Function],
                              "getActions": [Function],
                              "getState": [Function],
                              "replaceReducer": [Function],
                              "subscribe": [Function],
                            },
                            "unsubscribe": [Function],
                          },
                          "store": Object {
                            "clearActions": [Function],
                            "dispatch": [Function],
                            "getActions": [Function],
                            "getState": [Function],
                            "replaceReducer": [Function],
                            "subscribe": [Function],
                          },
                          "unsubscribe": [Function],
                        }
                      }
                    >
                      <News
                        __progress__="LOADING"
                        dispatch={[Function]}
                        loading={true}
                        net="MainNet"
                        networkId={null}
                        store={
                          Object {
                            "clearActions": [Function],
                            "dispatch": [Function],
                            "getActions": [Function],
                            "getState": [Function],
                            "replaceReducer": [Function],
                            "subscribe": [Function],
                          }
                        }
                        storeSubscription={
                          t {
                            "listeners": Object {
                              "clear": [Function],
                              "get": [Function],
                              "notify": [Function],
                              "subscribe": [Function],
                            },
                            "onStateChange": [Function],
                            "parentSub": t {
                              "listeners": Object {
                                "clear": [Function],
                                "get": [Function],
                                "notify": [Function],
                                "subscribe": [Function],
                              },
                              "onStateChange": [Function],
                              "parentSub": t {
                                "listeners": Object {
                                  "clear": [Function],
                                  "get": [Function],
                                  "notify": [Function],
                                  "subscribe": [Function],
                                },
                                "onStateChange": [Function],
                                "parentSub": undefined,
                                "store": Object {
                                  "clearActions": [Function],
                                  "dispatch": [Function],
                                  "getActions": [Function],
                                  "getState": [Function],
                                  "replaceReducer": [Function],
                                  "subscribe": [Function],
                                },
                                "unsubscribe": [Function],
                              },
                              "store": Object {
                                "clearActions": [Function],
                                "dispatch": [Function],
                                "getActions": [Function],
                                "getState": [Function],
                                "replaceReducer": [Function],
                                "subscribe": [Function],
                              },
                              "unsubscribe": [Function],
                            },
                            "store": Object {
                              "clearActions": [Function],
                              "dispatch": [Function],
                              "getActions": [Function],
                              "getState": [Function],
                              "replaceReducer": [Function],
                              "subscribe": [Function],
                            },
                            "unsubscribe": [Function],
                          }
                        }
                      >
                        <div
                          className="newsContainer"
                        >
                          <Connect(withData(withProps(HeaderBar)))
                            label="News"
                            net="MainNet"
                            networkId={null}
                          >
                            <withData(withProps(HeaderBar))
                              dispatch={[Function]}
                              label="News"
                              net="MainNet"
                              networkId={null}
                            >
                              <HeaderBar
                                dispatch={[Function]}
                                label="News"
                                net="MainNet"
                                networkId={null}
                              >
                                <div
                                  className="currentNetwork"
                                >
                                  MainNet
                                </div>
                                <div
                                  className="headerBar"
                                >
                                  <h3>
                                     
                                    News
                                  </h3>
                                </div>
                              </HeaderBar>
                            </withData(withProps(HeaderBar))>
                          </Connect(withData(withProps(HeaderBar)))>
                          <Panel
                            className="newsPanel"
                            onScroll={[Function]}
                            renderHeader={null}
                          >
                            <div
                              className="panel newsPanel"
                              onScroll={[Function]}
                            >
                              <Content
                                className="content"
                              >
                                <div
                                  className="content content"
                                >
                                  <Loader>
                                    <div
                                      className="loader"
                                    >
                                      <div />
                                      <div
                                        className="rect2"
                                      />
                                      <div
                                        className="rect3"
                                      />
                                      <div
                                        className="rect4"
                                      />
                                      <div
                                        className="rect5"
                                      />
                                    </div>
                                  </Loader>
                                </div>
                              </Content>
                            </div>
                          </Panel>
                        </div>
                      </News>
                    </withData(withProps(News))>
                  </Connect(withData(withProps(News)))>
                </withProgress(withProps(Connect(withData(withProps(News)))))>
              </withData(Connect(withProgress(withProps(Connect(withData(withProps(News)))))))>
            </Connect(withData(Connect(withProgress(withProps(Connect(withData(withProps(News))))))))>
          </withoutProps(Connect(withData(Connect(withProgress(withProps(Connect(withData(withProps(News)))))))))>
        </withCall(Connect(withData(Connect(withProgress(withProps(Connect(withData(withProps(News)))))))))>
      </Connect(withCall(Connect(withData(Connect(withProgress(withProps(Connect(withData(withProps(News))))))))))>
    </Router>
  </MemoryRouter>
</Provider>
`;
